backend/tool_instance/serializers.py

```diff
@@ -68,6 +68,11 @@ def create(self, validated_data: dict[str, Any]) -> Any:
             raise ValidationError(f"Workflow with ID {workflow_id} does not exist.")
         validated_data[TIKey.WORKFLOW] = workflow
 
+        if workflow.workflow_tool.count() > 0:
+            raise ValidationError(
+                f"Workflow with ID {workflow_id} can't have more than one tool."
+            )
```

Comment:
@johnyrahul I believe we need to have this check only for prompt studio exported tools - it'll be good to check with @hari-kuriakose

Reply:
Validated this with @shuveb 
cc: @hari-kuriakose @chandrasekharan-zipstack